tests/sla_tests.js

```diff
@@ -0,0 +1,212 @@
+var asyncDelay = 100;
+
+/**
+ * Returns a mock client that houses the bridges and can create both bridges
+ * and channels.
+ */
+
+var getMockClient = function() {
+  if(mockClient) {
+    return mockClient;
+  }
+  var Client = function() {
+    this.bridges = {
+      list: function(cb) {
+        cb(null, bridges);
+        if(bridges.length !== 0) {
+          usingExisting = true;
+        }
+        return bridges;
+      },
+      create: function(param, cb) {
+        bridges.push(getMockBridge(param));
+        cb(null, bridges[bridges.length-1]);
+      }
+    };
+    this.Channel= function() {
+      return getMockChannel();
+    };
+  };
+  mockClient = new Client();
+  return mockClient;
+};
+
+/**
+ * Returns a mock bridge that has a bridge_type and id, and can add channels
+ * to its array of channels. Emits a ChannelEnteredBridge event
+ */
+
+var getMockBridge = function(param) {
+  var Bridge = function(param) {
+    this['bridge_type'] = param.type;
+    if (this['bridge_type'] === 'mixing') {
+      isMixing = true;
+    }
+    this.id = bridgeId.toString();
+    bridgeId += 1;
+    this.addChannel= function(input, cb) {
+      //Here channels and bridges should have the same ID number
+      var channel = channels.filter(function(testChan) {
+        return testChan.id === input.id;
+      })[0];
+      bridgeChannels.push(channel);
+      cb(null);
+      this.emit('ChannelEnteredBridge', {bridge: this.id,
+        channel: channel.id}, {bridge: this, channel: channel});
+    };
+  };
+  util.inherits(Bridge, Emitter);
+  var mockBridge = new Bridge(param);
+  return mockBridge;
+};
+
+/**
+ * Returns a mock channel that has an id, and can use originate to add a channel
+ * to the array of total channels. Emits a StasisStart event when originating.
+ */
+
+var getMockChannel = function() {
+  var Channel = function() {
+    this.id = channelId.toString();
+    channelId += 1;
+    this.originate = function(input, cb) {
+      var self = this;
+      setTimeout(function() {
+        if(validEndpoint === input.endpoint) {
+          self.emit('StasisStart', {channel: {id: self.id}}, {channel: self});
+          pastStasis = true;
+        }
+      }, asyncDelay);
+    };
+  };
+  util.inherits(Channel, Emitter);
+  var mockChannel = new Channel();
+  channels.push(mockChannel);
+  return mockChannel;
+};
+
+describe('SLA Bridge and Channel Tester', function() {
+
+  afterEach(function(done) {
+    bridges = [];
+    usingExisting = false;
+    validEndpoint = 'SIP/phone';
+    pastStasis = false;
+    channels = [];
+    bridgeChannels = [];
+    isMixing = false;
+    done();
+  });
+
+  it('should create a bridge when there isn\'t one', function(done) {
+    var client = getMockClient();
+    var sla = require('../lib/sla.js')(client).done();
+
+    bridgeChecking();
+    function bridgeChecking() {
+      setTimeout(function() {
+        if(channels.length !== 0 && usingExisting === false &&
```

Comment:
since if statements already test for truth, you can just use  !usingExisting instead of usingExisting === false and isMixing instead of isMixing === true. Also, because 0 evaluates to false, you can just use channels.length instead of channels.length !== 0 but you can leave that if you want to make it explicit.